src/browser/api/webcontents.ts

```diff
@@ -1,9 +1,63 @@
 import * as url from 'url';
 import { app } from 'electron';
+import { Identity } from '../../shapes';
 import ofEvents from '../of_events';
-import route from '../../common/route';
+import route, { WindowRoute } from '../../common/route';
 import { InjectableContext, EntityType } from '../../shapes';
 
+export function hookWebContentsEvents(webContents: Electron.WebContents, { uuid, name }: Identity, topic: string, routeFunc: WindowRoute) {
+    webContents.on('did-get-response-details', (e,
+        status,
+        newUrl,
+        originalUrl,
+        httpResponseCode,
+        requestMethod,
+        referrer,
+        headers,
+        resourceType
+    ) => {
+        const type = 'resource-response-received';
+
+        const payload = {
```

Comment:
Looks great, the only big difference I can see is that instead of `Object.assign`ing the payload, we're picking out specific items to include in the payload. On a quick glance over the code, it seems like this is fine, so I'm assuming you've considered this, yeah?

Reply:
yep, this new format plays nicer with typescript and is more readable. This used to be set in a call that was modifying the payload